lib/matchers/matchers.rb

```
@@ -338,3 +338,164 @@ def format_expectation(negate)
     end
   end
 end
+
+# The following are required for the PAM resource
+#
+# Copyright 2019 - Onyx Point, Inc.
+#
+# TODO explain somewhere that :all_with_args, :all_without_args, :all_with_integer_arg
+# will cause match_pam_rule to return true when there are no potential matches
+RSpec::Matchers.define :match_pam_rule do |expected|
+  def matching_integer_arg? (line)
+    line.module_arguments.any? do |arg|
+      key, value = arg.split('=')
+
+      value && (@args[:key] == key) && value.match?(/^-?\d+$/) &&
+        value.to_i.send(@args[:operator].to_sym, @args[:value])
+    end
+  end
+
+  match do |actual|
+    case @args_type
+    when :all_with_args, :all_without_args, :all_with_integer_arg
+      retval = true
+    when :any_with_args, :any_with_integer_arg
+      retval = false
+    end
+
+    if [:all_with_integer_arg, :any_with_integer_arg].include? @args_type
+      unless Numeric.method_defined?(@args[:operator])
+        fail("Error: Operator '#{@args[:operator]}' is an invalid numeric comparison operator.")
```

Please raise an exception, and mention what it is - a PAM operator, otherwise this may be confusing.

---

`fail` == `raise`. I prefer `raise` because not everyone knows that... is there a specific exception class that should be raised tho? (I'm new—no clue)